benchmarks/python/test_matmul.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+from nvfuser import FusionDefinition, DataType
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from .core import run_benchmark, clear_cuda_cache
+import torch
+
+import csv
+import os
+
+
+def matmul_fusion(fd: FusionDefinition, dtype: DataType) -> None:
+    # Decide contiguity based on layout
+    a = fd.define_tensor(
+        shape=[-1, -1], contiguity=[True, True], dtype=dtype, is_cpu=False
+    )
+    b = fd.define_tensor(
+        shape=[-1, -1], contiguity=[True, True], dtype=dtype, is_cpu=False
+    )
+    out = fd.ops.matmul(a, b)
+    fd.add_output(out)
+
+
+precision_types = {"H": torch.float16, "S": torch.float32, "T": torch.bfloat16}
+
+
+def load_matmul_problems():
+    with open(os.path.join(os.path.dirname(__file__), "matmul_problems.csv"), "r") as f:
+        reader = csv.reader(f)
+        next(reader, None)  # skip header row
+        return list((int(m), int(n), int(k), layout) for m, n, k, layout in reader)
```

Comment:
We could also add batch size to this.

Reply:
It would also be interesting to see if there's a perf difference between a 2D matmul and a bmm with size 1 in our system. Of course as far as parametrization, we also do not particularly want to have 32,000 benchmarks run automatically necessarily.

Reply:
Batched GEMMs are not supported by the API rn. We only accept 2D matmuls

Reply:
I can open an issue to start development for this,

Reply:
Created Issue #2092

Reply:
Sounds good. We should remember to enable this once it's supported. Thanks for opening the issue.